[18.0][MIG] stock_restrict_lot: Migration to 18.0

mav-adhoc opened the pull request.

florian-dacosta committed:
Add module stock_restrict_lot

florian-dacosta committed:
Add module stock_restrict_lot

florian-dacosta committed:
Avoid copying restrict_lot_id field on stock move unless it is a split

florian-dacosta committed:
Avoid copying restrict_lot_id field on stock move unless it is a split

oca-travis committed:
[UPD] Update stock_restrict_lot.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] icon.png

newtratip committed:
[14.0][FIX] stock_restrict_lot

newtratip committed:
[14.0][FIX] stock_restrict_lot

stock_restrict_lot 14.0.1.1.1

OCA-git-bot committed:
stock_restrict_lot 14.0.1.1.1

bizzappdev committed:
[IMP]stock_restrict_lot: migrate stock_restrict_lot module

[UPD] Update stock_restrict_lot.pot

[UPD] README.rst

petrus-v committed:
[FIX] stock_restrict_lot: do not merge stock move with different restrict_lot_id

Odoo tries to avoid extra stock.move while generating stock.picking from
pull/push rules. We must tell odoo to not merge stock.move if they are restricted
different lot.

To reproduce this behaviour using sale_order_lot_selection you have to:
* setup inventory with multi-step routes
* setup warehouse with two steps outgoing shipments
* receive products with different lot
* create a sale order with different quantities per lot
* validate the sale order

Before this PR stock.move in the internal stock.picking pick order are merged

After this PR we have one line per selected lot in internal stock.picking pick order

stock_restrict_lot 15.0.0.0.2

florian-dacosta committed:
[MIG] Migrate stock_restrict_lot to v16

Commit:
[UPD] Update stock_restrict_lot.pot

OCA-git-bot committed:
[UPD] README.rst

florian-dacosta committed:
[IMP][stock_restrict_lot] Add lot in move list views

mt-software-de committed:
[14.0][IMP] compute product's outgoing and incoming qty by restrict_lot_id

Commit:
[UPD] Update stock_restrict_lot.pot

OCA-git-bot committed:
stock_restrict_lot 16.0.1.1.0

badbole committed:
Added translation using Weblate (Croatian)

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (6 of 6 strings)

Translation: stock-logistics-workflow-16.0/stock-logistics-workflow-16.0-stock_restrict_lot
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-16-0/stock-logistics-workflow-16-0-stock_restrict_lot/hr/

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-workflow-16.0/stock-logistics-workflow-16.0-stock_restrict_lot
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-16-0/stock-logistics-workflow-16-0-stock_restrict_lot/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (6 of 6 strings)

Translation: stock-logistics-workflow-16.0/stock-logistics-workflow-16.0-stock_restrict_lot
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-16-0/stock-logistics-workflow-16-0-stock_restrict_lot/it/

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (6 of 6 strings)

Translation: stock-logistics-workflow-16.0/stock-logistics-workflow-16.0-stock_restrict_lot
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-16-0/stock-logistics-workflow-16-0-stock_restrict_lot/es/

HekkiMelody committed:
[IMP] stock_restrict_lot: validate lot moved is correct

jcadhoc committed:
[IMP] stock_restrict_lot: pre-commit auto fixes

jcadhoc committed:
[MIG] stock_restrict_lot: Migration to 17.0

Commit:
[UPD] Update stock_restrict_lot.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-workflow-17.0/stock-logistics-workflow-17.0-stock_restrict_lot
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-17-0/stock-logistics-workflow-17-0-stock_restrict_lot/

grindtildeath committed:
[IMP] stock_restrict_lot: Add index to restrict_lot_id

Commit:
[UPD] Update stock_restrict_lot.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-workflow-17.0/stock-logistics-workflow-17.0-stock_restrict_lot
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-17-0/stock-logistics-workflow-17-0-stock_restrict_lot/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (8 of 8 strings)

Translation: stock-logistics-workflow-17.0/stock-logistics-workflow-17.0-stock_restrict_lot
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-17-0/stock-logistics-workflow-17-0-stock_restrict_lot/it/

mav-adhoc committed:
[IMP] stock_restrict_lot: pre-commit auto fixes

mav-adhoc committed:
[MIG] stock_restrict_lot: Migration to 18.0

rousseldenis commented:
/ocabot migration stock_restrict_lot

rousseldenis commented on stock_restrict_lot/__manifest__.py:
```diff
@@ -1,7 +1,7 @@
     "name": "Stock Restrict Lot",
     "summary": "Base module that add back the concept of restrict lot on stock move",
-    "version": "17.0.1.1.0",
+    "version": "18.0.1.1.0",
```
Don't forget to reset trailing numbers too (`18.0.1.0.0`). Thanks

StefanRijnhart reviewed (changes requested):
Some tests from this module seem to be failing.

mav-adhoc commented:
@StefanRijnhart What should i do? Change the test? There is no change on the module's models

StefanRijnhart commented:
@mav-adhoc It must mean that the logic of the module no longer fits the exact processes in Odoo that it tries to change. This could be a very small detail, or it could mean that the Odoo process has been revised completely in the new edition. You'll have to study the changes in the way that Odoo handles lots between 17.0 and 18.0. You can start out by checking if all the methods that this module overrides still exist, and if their signatures are still the same.
Another strategy is to trace the tests in the old and new versions and see where the results start to diverge.

DavidJForgeFlow committed:
[REF] stock_restrict_lot: change AGPL-3 to LGPL-3

mav-adhoc commented:
@StefanRijnhart Working on it!

DavidJForgeFlow commented:
Hi! Please add the changes added in #1840, thanks!

StefanRijnhart committed:
[FIX] stock_restrict_lot: fix outgoing quantity in test

The outgoing quantity is different depending on whether mrp is loaded, as the
demo data of that module adds an outgoing quantity of 3.
In the test, use a copy of the demo product to ensure there are no preexisting
stock operations.

Fixes
```
odoo.addons.stock_restrict_lot.tests.test_restrict_lot: FAIL: TestRestrictLot.test_compute_quantites
Traceback (most recent call last):
  File "/__w/stock-logistics-workflow/stock-logistics-workflow/stock_restrict_lot/tests/test_restrict_lot.py", line 254, in test_compute_quantites
    self.assertEqual(product.outgoing_qty, 2)
AssertionError: 5.0 != 2
```

StefanRijnhart commented:
This one needs to be included as well: https://github.com/OCA/stock-logistics-workflow/pull/1859

rjaraspearhead commented:
@mav-adhoc, Can you review the tests to ensure they pass the actions?

henrybackman commented on stock_restrict_lot/tests/test_restrict_lot.py:
```diff
@@ -0,0 +1,304 @@
+        pickings = self.env["stock.picking"].search(
+            [("group_id", "=", procurement_group.id)]
+        )
+        self.assertEqual(len(pickings), 1)
```
Not sure why this has been changed from `2` to `1` when the test comment above still mentions creating two pickings

StefanRijnhart commented:
Superseded by https://github.com/OCA/stock-logistics-workflow/pull/1880
